templates/taxi/car_detail.html

```diff
@@ -12,6 +12,11 @@ <h1>
     </a>
   </h1>
   <p>Manufacturer: ({{ car.manufacturer.name }}, {{ car.manufacturer.country }})</p>
+  {% if car in user.cars.all %}
+      <a href="{% url 'taxi:car-assign_to_car' pk=car.id %}" class="btn btn-danger link-to-page">Delete me from this car</a>
+    {% else %}
+      <a href="{% url 'taxi:car-assign_to_car' pk=car.id %}" class="btn btn-secondary link-to-page">Assign me to this car</a>
```

Comment:
you use here backward relationship but did you fetched it from db? because otherwise here will be N+1 issue

Reply:
Use a form instead of hyperlinks to send value to CarDetailView